### PR TITLE
Fixed custom module importing on Windows

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
# What does this PR do?

The current implementation doesn't import custom modules on Windows. Windows use back slash to separate directory hierarchies and `os.path.join()` follows that convention, which is different from the `rfilename`s extracted to `filenames` before.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

- Pipelines:  @patrickvonplaten and @sayakpaul

### CODE REVIEW

File: src/diffusers/pipelines/pipeline_utils.py [hunk 1]
```diff
@@ -1688,7 +1688,7 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                 if module_candidate is None or not isinstance(module_candidate, str):
                     continue
 
-                candidate_file = os.path.join(component, module_candidate + ".py")
+                candidate_file = f"{component}/{module_candidate}.py"
```
Comment (sayakpaul) [inline src/diffusers/pipelines/pipeline_utils.py:1691]: `os.path.join()` should automatically adapt to the directory separator conventions of the operating system it's running on. So, not fully sure here.

Comment (PENGUINLIONG) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: The problem is that paths in `filenames` are not using backslashes.

Comment (PENGUINLIONG) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: even on windows. I'm using win10 22H2

Comment (sayakpaul) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: What are they using? That's strange.

Comment (PENGUINLIONG) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: forward slash actually. I'm not sure if it's rfilename's side effect

Comment (PENGUINLIONG) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: https://github.com/huggingface/diffusers/blob/6f58f2f979b572479fcea57714156a1b9aec122c/src/diffusers/pipelines/pipeline_utils.py#L1677 this one

Comment (sayakpaul) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: Pinging @Wauplin maybe.

Comment (Wauplin) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: Yes @PENGUINLIONG is right here. `os.path.join` is useful when dealing with local filesystem. Here we want to compute a filepath on the Hub which uses the unix convention (e.g. `path/to/file.txt`). Good catch!
(but I wonder why we did not realize before though... :thinking:).

---

(nit) I would add a small comment for the future ourselves here.

```suggestion
                # We compute candidate file path on the Hub. Do not use `os.path.join`. 
                candidate_file = f"{component}/{module_candidate}.py"
```

Comment (sayakpaul) [reply src/diffusers/pipelines/pipeline_utils.py:1691]: Thanks, Lucain! The custom module component here is fairly new, probably that's why.

Comment (patrickvonplaten) [reply src/diffusers/pipelines/pipeline_utils.py:1692]: Could we maybe use https://docs.python.org/3/library/os.path.html#os.path.normpath

**Pull request merged**